core/lib/types/src/snapshots.rs

```diff
@@ -71,3 +71,10 @@ pub struct SnapshotFactoryDependencies {
 pub struct SnapshotFactoryDependency {
     pub bytecode: Vec<u8>,
 }
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct AppliedSnapshotStatus {
+    pub l1_batch_number: L1BatchNumber,
+    pub is_finished: bool,
+    pub last_finished_chunk_id: Option<u64>,
```

Comment:
I think here these columns would be better replaced with an enum to more accurately model the snapshot lifecycle; something like:

```rust
pub enum SnapshotRecoveryState {
    NotStarted,
    Pending { last_finished_chunk_id: u64 },
    Finished,
}
```

Reply:
It may be required to introduce more sophisticates progress tracking, but I'd try to delay making this change as long as possible, when we are sure this indeed is needed